go/actions/test/action.yml

```diff
@@ -0,0 +1,80 @@
+name: Test go extractor
+description: Run build, QL tests and optionally basic code sanity checks (formatting and generation)
+inputs:
+  run-code-checks:
+    description: Whether to run formatting, code and qhelp generation checks
+    required: false
+    default: false
+runs:
+  using: composite
+  steps:
+    - name: Get go version
+      shell: bash
+      run: |
+        (
+          echo -n "GO_VERSION="
+          bazel run @rules_go//go -- version | sed 's/go version go\(.*\) .*/\1/'
+        ) | tee -a "$GITHUB_ENV"
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ env.GO_VERSION }}
+        cache: false
+      id: go
```

Comment:
Thanks for the points you raise.

Ideally I would like to arrive at a point where the go version is stated in just one point, and not repeated here or in the internal repo. I thought the source of truth could be the line in `MODULE.bazel`, but I haven't considered the contents of `go.mod`. I'll experiment with that a bit. I'm not sure `go.mod` will have any effect any longer on the go version if we build with bazel.

We can indeed be more lenient on the patch level.

Reply:
If we want to have a single source of truth for the Go version (and, as I noted, I am not sure whether we do as we might want to distinguish between the version that the extractor is built with and the one that is in the environment for the autobuilder/extractor to use when running tests), then the thing to do here would probably be to specify `GOTOOLCHAIN=local` for the call to `go version`. That (should) guarantee that `go version` will report the version of the installed toolchain (as specified in `MODULE.bazel`) rather than e.g. the one in `go.mod`.

Reply:
good point about distinguishing between build version and runtime version (and about the `GOTOOLCHAIN` setting). I'll mull it over.

Reply:
as far as I could test, `go version` always prints the runtime version, ignoring the minimal go version writting in `go.mod`.

Right now, I'm thinking of:
* specifying the go version to build with in `MODULE.bazel`
* that is somewhat independent on what is written in `go.mod` and `go.work` files which is the declared minimal version the go extractor can be built with. I don't have a definitive proposal how to deal with that, other than keeping doing what we were doing and update it independently
* specifying the go version with which we intend to run tests in a `go/test_runtime_version` or something similar, and providing a local `setup-go` action wrapping `actions/setup-go` and reading the default version spec from this file. Another similar approach, without the wrapper action: write somewhere a `go.work` file (maybe in `go/ql/test`?) and provide that to `actions/setup-go`'s [go-version-file](https://github.com/actions/setup-go/blob/cdcb36043654635271a94b9a6d1392de5bb323a7/action.yml#L7-L8) input.

This way no change would be required in the internal repo when updating the go version. In any case, this is meat for the other PR.

Reply:
I see! thanks for the explanation. Though I guess that applies when running `go version` within the directory containing `go.mod`/`go.work` (which wouldn't be the case in this specific case.

So if I'm understanding you, you'd prefer to keep things as they are as far as declaring the go runtime version for tests.

The build version will however need to be declared separately in the `MODULE.bazel` file. But hopefully that will actually make it easier to test against different Go versions, because the version specified in `setup-go` won't affect the build itself any more.

Reply:
> Though I guess that applies when running go version within the directory containing go.mod/go.work (which wouldn't be the case in this specific case).

That's a good point. If `go version` is run in a directory without those files, it should just always print the environment version. (Though, if it's not too hard, adding `GOTOOLCHAIN=local` may still be nice to preserve our sanity if the Go folks start scanning subdirectories, like they are for some commands, or we put a `go.work` file in the root of the repository, etc.)

> So if I'm understanding you, you'd prefer to keep things as they are as far as declaring the go runtime version for tests.

I think that makes sense for us. 

> The build version will however need to be declared separately in the `MODULE.bazel` file. But hopefully that will actually make it easier to test against different Go versions, because the version specified in `setup-go` won't affect the build itself any more.

That also sounds good.

Reply:
I'll resolve this conversation in this PR, as this will be implemented over in the other sibling PR. Thanks for the comments!